csharp/src/Ice/OutgoingFrame.cs

```diff
@@ -197,10 +316,63 @@ public CompressionResult CompressPayload()
             _compressionMinSize = compressionMinSize;
         }
 
-        internal void Finish(OutputStream.Position encapsulationEnd)
+        private OutputStream WriteBinaryContextEntryKey(int key)
```

Comment:
I would rather not create a new OuputStream for each entry - especially not for the ReadOnlySpan entries. And maybe we could cache a single OutputStream for all binary context entries that are not ReadOnlySpan<byte>? It's not like we allow writing binary context entries concurrently.

Reply:
Replaced with a cached OutputStream as mentioned above